pkg/controller/composite/controller.go

```diff
@@ -596,6 +597,15 @@ func (pc *parentController) syncParentObject(parent *unstructured.Unstructured)
 	// Update parent status.
 	// We'll want to make sure this happens after manageChildren once we support observedGeneration.
 	if _, err := pc.updateParentStatus(parent, syncResult.Status); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Swallow the error since there's no point retrying if the parent is gone.
+			pc.logger.V(4).Info("Parent object has been deleted", "parent_kind", pc.parentResource.Kind, "object", klog.KRef(parent.GetNamespace(), parent.GetName()))
+			return nil
+		} else if apierrors.IsConflict(err) {
+			// it is possible that the object was modified after this sync was started, ignore conflict since we will reconcile again
+			pc.logger.V(4).Info("Parent ignoring update due to outdated resourceVersion", "parent_kind", pc.parentResource.Kind, "object", klog.KRef(parent.GetNamespace(), parent.GetName()))
+			return nil
```

Comment:
if we return here `nil` how metacontroller will know that it should retry ?

Reply:
If my understanding is correct, for this case, if we get a 409 then the current resource in the cluster differs from what we read from the cache. So, there is no need to retry explicitly here because the watch on the resource in the sharedinformer will see the change and trigger an update in the cache/workqueue and it will get retried with the latest version in the cluster.